### PR TITLE
Migrate to new versions of modules and ES6.

### LINKED ISSUE #15: Document reply.proxy
### `reply.proxy(options)`

Proxies the request to an upstream endpoint where:
- `options` - an object including the same keys and restrictions defined by the
 [route `proxy` handler options](#route.config.proxy).

No return value.

The [response flow control rules](#flow-control) **do not** apply.

```js
var handler = function (request, reply) {

    return reply.proxy({ host: 'example.com', port: 80, protocol: 'http' });
};
```

### PR DESCRIPTION
**Pull request by osukaa**
I'm upgrading the versions for all the Hapi modules plus moving to ES6 with the support for Node 4+. This also closes #15

### CODE REVIEW

**Comment by mark-bradshaw** [comment]
This may sound strange, but switch lines 125 - 127 to:
```js
if (settings.xforward &&
                request.info.remotePort &&
                request.info.remoteAddress ) {
```

File: .travis.yml [hunk 1]
```diff
@@ -1,7 +1,8 @@
 language: node_js
 
 node_js:
-  - 0.10
-  - 0.12
-  - iojs
+  - 4.0
```
Comment (Marsup) [inline .travis.yml]: Quotes around all version numbers.

File: package.json [hunk 2]
```diff
@@ -12,18 +12,19 @@
...
   "engines": {
-    "node": ">=0.10.32"
+    "node": ">=4.0.0"
   },
   "dependencies": {
-    "boom": "2.x.x",
-    "hoek": "2.x.x",
-    "joi": "6.x.x",
-    "wreck": "6.x.x"
+    "boom": "^2.10.1",
```
Comment (Marsup) [inline package.json]: v.x.x format

File: README.md [hunk 3]
```diff
@@ -76,6 +74,23 @@ The proxy handler object has the following properties:
...
+Proxies the request to an upstream endpoint where:
+- `options` - an object including the same keys and restrictions defined by the
+ [route `proxy` handler options](#route.config.proxy).
+
+No return value.
+
+The [response flow control rules](#flow-control) **do not** apply.
+
+```js
+var handler = function (request, reply) {
```
Comment (Marsup) [inline README.md]: ES6 even in examples -> const

**Pull request merged**